### PR TITLE
Update debug

### PR DESCRIPTION
**Pull request by joakim-hove**
Attempt to debug why clients restart so often

### CODE REVIEW

File: bin/fby_client [hunk 1]
```diff
@@ -58,8 +58,14 @@ class FbyRunner(object):
...
     def updateClient(cls, config):
-        new_config = config.downloadNew( )
-        if config.updateRequired( new_config ):
+        if config.updateRequired( ):
+
+            data = ""
+            for key in config.data:
+                data += "%12s  %s -- %s\n" % (key , config.data[key] , new_config.data[key])
+                
+            config.logMessage("Restarting - git_follow: %s" % config.getGitFollow , long_msg = data)
```
Comment (pgdr) [inline bin/fby_client]: getGitFollow is a function?

Comment (pgdr) [reply bin/fby_client]: Can't you take `long_msg = json.dumps(config.data)`?

File: lib/device_config.py [hunk 2]
```diff
@@ -79,11 +79,12 @@ def getGitFollow(self):
...
 
-    def updateRequired(self, other):
+    def updateRequired(self):
         if self.getGitFollow( ):
             return True
-
-        return self != other
+        else:
+            new_config = self.downloadNew( )
+            return self != new_config
```
Comment (pgdr) [inline lib/device_config.py]: Why not just test for the git_ref?

``` python
if not hasattr(new_config, 'data'):
    return False
return self.data['git_ref'] != new_config.data['git_ref']
```

File: bin/fby_client (hunk 1 above)
Comment (joakim-hove) [reply bin/fby_client]: > `getGitFollow` is a function?

Yes it is.

> Can't you take `long_msg = json.dumps(config.data)?`

Yes - good idea.

File: lib/device_config.py (hunk 2 above)
Comment (joakim-hove) [reply lib/device_config.py]: Yes - I guess that would be just as good.

**Review by pgdr — APPROVED** [review]
This looks good.  Please merge and release or retag!

**Pull request merged**